llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -289,6 +289,43 @@ class OpLowerer {
     });
   }
 
+  void lowerTypedBufferStore(Function &F) {
+    IRBuilder<> &IRB = OpBuilder.getIRB();
+    Type *Int8Ty = IRB.getInt8Ty();
+    Type *Int32Ty = IRB.getInt32Ty();
+
+    replaceFunction(F, [&](CallInst *CI) -> Error {
```

Comment:
why does this need to be a lambda?

Reply:
It doesn't *need* to be - it just seemed more ergonomic than coming up with a name for it and spelling it elsewhere. Do you have a different suggestion?

Reply:
The thing that's throwing me here is that we have a bunch of `replaceFunction` Lambda's and we have a replaceFunction method in `OpLowerer` and what I'm wondering is why do we do methods sometimes and lambdas others times.  further can `lowerTypedBufferLoad` share anything with `lowerTypedBufferStore` `replaceFunction ` if they weren't lambdas?

Reply:
We aren't doing methods sometimes and lambdas others, we have a method on the `OpLowerer` called `replaceFunction` that takes a lambda `ReplaceCall` that replaces an individual call, and then each kind of function we want to replace specializes behaviour through that. The idea here is that we can share the boilerplate for looping through the users, constructing an error message, and erasing the function itself once it's been replaced.

The other "obvious" way to structure this would be to have some kind of "FunctionReplacer" class that does the boilerplate and then subclass it for each of what are currently lambdas. I don't like this as much since it makes it harder to access the shared state of the OpLowerer in that model.

Reply:
It would help with clarity if `OpLowerer::replaceFunction`  and the specialized replace function behaviors had different names.

Reply:
I don't understand what you mean. The lambdas don't have names, they're arguments to `replaceFunction`.